app/controllers/nonsipp/shares/HowManySharesController.scala

```diff
@@ -76,7 +77,8 @@ class HowManySharesController @Inject()(
               Future.successful(
                 BadRequest(
                   view(
-                    HowManySharesController.viewModel(srn, index, nameOfSharesCompany, mode, formWithErrors)
+                    formWithErrors,
+                    HowManySharesController.viewModel(srn, index, nameOfSharesCompany, mode, form)
```

Comment:
Does it matter that `form` is different from `formWithErrors` here (and probs in some other places too)?